### PR TITLE
Feature/432 show all layers in list

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [HMW-432](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-432)

## Main Changes:
* Switched to showing all layers (that don't have a list mode of "hide") in the layer list on every tab of the _Community_ page, rather than selectively removing some.
* Moved all layers, layer visibility state, and layer error state from the `locationSearch` context into the `Layers` context.
  * There are a good amount of keys in the context that are unused. The `errored` state object was initially going to be used in more places, but we decided not to update it based on service calls. However, some layers do still use it, and I don't think it hurts to leave the keys in place in case we want to use them in the future.
* Stopped checking for service errors when determining layer visibility. If there is a service error, the layer will simply not have any features.
* Fixed a dependency bug in the Surrounding Features widget, where changing a different surrounding layer's visibility would cause a layer's service call to abort.

## Steps To Test:
1. Navigate around the various tabs of the _Community_ page and verify that layers are not removed from the layer list or the Surrounding Features widget as tabs are changed.
2. The default layer visibility should be the same as in production for each tab.
3. One exception to the layer list inclusion is on the _Identified Issues_ tab: since the `issuesLayer` only displays a subset of the features in the `waterbodyLayer`, only the former is shown on this tab.

### CODE REVIEW

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Looks good and works well!

Here are some things that I noticed:
* The `State Watershed Health Index` layer never seems to show up in the layer list, even if I have the layer turned on from the Protect panel.
* The `All Waterbodies` layer is sometimes disabled. Sometimes I can get it re-enabled by changing my zoom level.
  * On the monitoring report page, the `All Waterbodies` layer is stuck disabled and changing zoom doesn't fix it.

File: app/client/src/utils/hooks/index.ts [hunk 1]
```diff
@@ -1352,7 +1372,7 @@ function useSharedLayers() {
       url: services.data.stateBoundaries,
       title: 'State',
       sublayers: [{ id: 0 }],
-      listMode: 'hide',
+      listMode: 'show',
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/client/src/utils/hooks/index.ts]: Yes, good call!

**Review by cschwinderg — APPROVED** [review]
Thanks for making those changes!

I noticed an issue where I couldn't save the upstream watershed. The issue was because of changing the id of the layer. I went ahead and commited that change.

**Comment by cschwinderg** [comment]
@maxdiebold-erg Feel free to merge this in, when you're ready.

**Pull request merged**